features/step_definitions/evidence_hub/search_steps.rb

```diff
@@ -31,6 +31,23 @@
   expect(evidence_summaries_page.search_results.size).to be(records_size.to_i)
 end
 
+Then('I should see the {string} evidence summary with data types as') do |result_number, table|
+  search_result = evidence_summaries_page.search_results.send(result_number)
+
+  icon_classes = {
+    'tick' => 'evidence-types__item-icon--tick',
+    'cross' => 'evidence-types__item-icon--cross'
+  }
+
+  table.rows.each do |row|
+    field_name = row[0]
+    icon = row[1]
+    data_type_field = search_result.send(field_name)
+
+    expect(data_type_field['class']).to include(icon_classes[icon])
+  end
+end
```

Comment:
I know that we paired on this but I still quite not happy with the test feedback.

If this fails this will be returned: `expected "evidence-types__item-icon evidence-types__item-icon--tick svg-icon svg-icon--evidence-tick" to include "evidence-types__item-icon--cross"`

This message is not so intuitive for any developer that will face this.

A suggestion will be customise the message. I am not proud of this code but I have some sort of idea that worked (just need more refactor to the code). But here it is:

```ruby
table.rows.each do |row|
    field_name = row[0]
    icon = row[1]

    data_type_field = search_result.send(field_name)
    data_type_field_html_class = data_type_field['class']
    icon_class_value = data_type_field_html_class[/tick|cross/] # get the tick value or the cross value from the class

    expect(data_type_field_html_class).to include(icon_classes[icon]),
      "expected to data type '#{field_name}' to be '#{icon}'. Got: #{icon_class_value}"
  end
```

The result: `expected to data type 'qualitative' to be 'tick'. Got: cross` which I think it get's a better feedback. @margOnline and @willhall88  wdyt?

Reply:
Or you can take the opportunity that you grep the "tick" or the "cross" from  the html class and have a simple expectation `expect(icon_class_value).to eq(icon)` I am okay with either solutions.